nameko/timer.py

```diff
@@ -49,26 +51,29 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-
-        sleep_time = self.interval
+        sleep_time = 0 if self.eager else self.interval
 
         while True:
             # sleep for `sleep_time`, unless `should_stop` fires, in which
             # case we leave the while loop and stop entirely
-            with Timeout(sleep_time, exception=False):
-                self.should_stop.wait()
-                break
+            time.sleep(sleep_time)
 
             start = time.time()
 
             self.handle_timer_tick()
 
+            self.event.wait()
+            self.event.reset()
+
             elapsed_time = (time.time() - start)
 
             # next time, sleep however long is left of our interval, taking
             # off the time we took to run
             sleep_time = max(self.interval - elapsed_time, 0)
 
+            if self.should_stop:
```

Comment:
Not a big deal but this check can go higher in the loop. There's no need to reset the event or work out the new `sleep_time` if you're going to exit the loop.